### PR TITLE
Delete customer list entries helper

### PR DESCRIPTION
**Pull request by skeletonkey**
TL:DR; remove list entries using line numbers instead of some type of string matching.

I was having issues figuring out how to remove entries with the current remove logic.  Now that I've dug into the code I get how it works, however, it seems something easy to forget.  To make removing entries a no-brainer I added logic to delete entries by line number.  This make working with larger lists a pain, but I figure that if that is the case you're probably editing the lists directly with your favorite editor.

I went back and forth on adding new menu entries, however, I felt that extra entries would confuse the UI.  Thus I replaced the current remove functionality.  Let me know if you feel that the old functionality is still desired and I can add those back in with new entries for this new way.

### CODE REVIEW

**Comment by VR51** [comment]
Looks good to me. My intention when I rebuilt HF was to integrate custom list management with a text editor. I've processed this merge request. Thanks @skeletonkey.

**Comment by skeletonkey** [comment]
I finally read your dev notes in the script - it's amazing what you learn
when you actually read :P

Thank you for accepting the two merge requests as it's really helping me
properly curating my whitelists. I like the idea that you have about being
able to edit the files with an editor but this last changes will be a nice
stop gap.

My next focus will be to add 'custom DNS settings' - this would go under
Option 2 of the main menu.  I'm a programmer and sometime I need the
ability to point a URL past a proxy to a specific node.  I'm playing with
how best to do this, but the ultimate solution would be to have a menu
(like Option 1 - List Management) where you can have named lists that will
be prepended to the hosts file.

Another idea that I'd like to tinker with is using an HTTP HEAD request to
compare each list size to a cached version to stop downloading the same
list over and over again - thereby speeding up the rebuilding of the list.
I'm not sure that this will actually have any speed implications since most
lists are updated regularly.

I'm mentioning these two points so that you know where I'm headed and I'm
not stomping on work that you are currently doing!

Thanks for script - it's fantastic!

Erik

**Comment by VR51** [comment]
I'm happy for your input. Stomp away at your leisure.

Custom DNS pointer management is a good idea. I usually edit the hosts file directly to add them but I like the idea of adding this feature to Host Flash.

At the moment, any custom DNS rules near the top of a system's active hosts file are copied to the top of the host file on rebuild.

Note that around line 1104 the script tests the active hosts file for the headline comment **'# Host.? Flash'** (I've just noticed I forgot to update the test in the __disable function at line 1160 when I adjusted the heading text -- will update it later on). If we were to add an option to add custom IP --> host pointers it would need a comment above the set of custom pointers so that they can easily be extracted and restored during builds such that non managed stuff stays untouched.

I would add the option like so,

- Create a menu for Managed DNS Pointers (or some other title)
- Create a submenu for: view, add, delete, edit, en/dis-able pointers (comment/uncomment)
- Add new top level menu item to enable/disable all Managed DNS Pointers
- Set host flash to create a new config file of .hfmdns to store managed DNS pointers (see line 212 to see how the config files are currently added)
- Add new headline of '# Host Flash Managed DNS Pointers' to the hosts file
- On hosts file build, insert the content of .hfmdns below the heading '# Host Flash Managed DNS Pointers' but above the heading '# Host Flash Firewall'

Alternatively, replace the submenu with a call to a text editor to open .hfmdns for direct editing then we need only enable/disable the whole set after each edit.

I would shy away from copying the whole default hosts file head to (e.g) .hfmdns. Default host file content is occasionally changed during OS upgrades and this could be an issue if someone were to restore their home directly after an OS reinstall then ran Host Flash.
